plugins/server_links/ui/ServerLinks.component.ts

```diff
@@ -90,89 +104,98 @@ export default class ServerLinks extends StaticComponent {
+        }
+        if (pageCount === 1) {
+            buttonAmount = 0
+        }
+        const headerCfg: StaticHeaderOptions = this.header.options
+        let buttonsXml: string = ''
+        for (let i: number = 0; i < buttonAmount; i++) {
+            const action: string = ids[i] === undefined ? '' : `action="${ids[i]}"`
+            const icon: string = icons[i] === undefined ? '' : `<quad posn="${headerCfg.iconHorizontalPadding} ${-headerCfg.iconVerticalPadding} 4" 
       sizen="${headerCfg.iconWidth} ${headerCfg.iconHeight}" image="${icons[i]}" imagefocus="${iconsHover[i]}" ${action}/>`
-      buttonsXml += `<frame posn="${(config.width + config.margin) - (headerCfg.squareWidth + config.margin) * ((buttonAmount - i))} 0 1">
+            buttonsXml += `<frame posn="${(config.width + config.margin) - (headerCfg.squareWidth + config.margin) * ((buttonAmount - i))} 0 1">
         <quad posn="0 0 1" sizen="${headerCfg.squareWidth} ${headerCfg.height}" bgcolor="${headerCfg.textBackground}"/>
         ${icon}
       </frame>`
-    }
-    return `<quad posn="0 0 1" sizen="${headerCfg.squareWidth} ${headerCfg.height}" bgcolor="${headerCfg.textBackground}"/>
+        }
+        return `<quad posn="0 0 1" sizen="${headerCfg.squareWidth} ${headerCfg.height}" bgcolor="${headerCfg.textBackground}"/>
     <quad posn="${headerCfg.iconHorizontalPadding} ${-headerCfg.iconVerticalPadding} 4" sizen="${headerCfg.iconWidth} ${headerCfg.iconHeight}" image="${config.icon}"/> 
     <frame posn="${headerCfg.squareWidth + config.margin} 0 1">
     <quad posn="0 0 1" sizen="${config.width - (headerCfg.squareWidth + config.margin) * (1 + buttonAmount)} ${headerCfg.height}" bgcolor="${headerCfg.textBackground}"/>
-      ${leftAlignedText(config.title, config.width - (headerCfg.squareWidth + config.margin) * (1 + buttonAmount), headerCfg.height, { textScale: headerCfg.textScale, padding: headerCfg.horizontalPadding })}
+      ${leftAlignedText(config.title, config.width - (headerCfg.squareWidth + config.margin) * (1 + buttonAmount), headerCfg.height, {
+            textScale: headerCfg.textScale,
+            padding: headerCfg.horizontalPadding
+        })}
     </frame>
     ${buttonsXml}
     `
-  }
+    }
 
-  private constructEntry(ii: number, jj: number, ww: number, hh: number, startIndex: number): string {
-    const data = this.serverList[(ii + startIndex) % this.serverList.length]
-    const m = config.margin
-    const arr: GridCellFunction[] = [
-      (i, j, w, h): string => {
-        return `${this.icon(icons.name, iw, h)}
-        ${this.text(tm.utils.strip(data.name, false), w - iw - m, h, iw + m, config.iconBackground, true)}`
-      },
-      (i, j, w, h): string => {
-        const width = 7
-        return `${this.icon(icons.map, iw, h)}
+    private constructEntry(ii: number, jj: number, ww: number, hh: number, startIndex: number): string {
+        const data = this.serverList[(ii + startIndex) % this.serverList.length]
```

Comment:
Instead of solving servers being offline like this we could a) just not display them b) display them but gray them out and this is probably the best solution but it would require us to cache the server info somehow because its obv not available if the server is offline. Now since the servers are gonna be online 99% of the time its not that important so we can leave it like this and maybe fix it later